File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/SubCategoryDto.cs
using ProjectPos.Services.EntityDtos;
using ProjectPos.Data.Shared.Enums;

namespace ProjectPos.Services.DTOs
{
    public class SubCategoryDto : EntityDto<int>
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public Category? Category { get; set; }
    }
}

File: ProjectPos.FrontOffice.FrontEnd/src/proxy/interfaces/sub-category-dto.ts
import { EntityDto } from "../entity-dtos/entity-dto";
import { Category } from "../enums/category";
import { ProductDto } from "./product-dto";

export interface SubCategoryDto extends EntityDto<number> {
    name?: string;
    description?: string;
    category?: Category;
    products?: ProductDto[];
}
